cns/middlewares/k8sSwiftV2.go

```diff
@@ -135,150 +134,62 @@ func (m *K8sSWIFTv2Middleware) validateIPConfigsRequest(ctx context.Context, req
-		PodIPConfig: cns.IPSubnet{
-			IPAddress:    ip.String(),
-			PrefixLength: uint8(prefixSize),
-		},
-		MacAddress:        mtpnc.Status.MacAddress,
-		NICType:           cns.DelegatedVMNIC,
-		SkipDefaultRoutes: false,
-		// InterfaceName is empty for DelegatedVMNIC
-	}
-
-	return podIPInfo, nil
-}
 
-// setRoutes sets the routes for podIPInfo used in SWIFT V2 scenario.
-func (m *K8sSWIFTv2Middleware) setRoutes(podIPInfo *cns.PodIpInfo) error {
-	logger.Printf("[SWIFTv2Middleware] set routes for pod with nic type : %s", podIPInfo.NICType)
-	podIPInfo.Routes = []cns.Route{}
-	switch podIPInfo.NICType {
-	case cns.DelegatedVMNIC:
-		virtualGWRoute := cns.Route{
-			IPAddress: fmt.Sprintf("%s/%d", virtualGW, prefixLength),
-		}
-		// default route via SWIFT v2 interface
-		route := cns.Route{
-			IPAddress:        "0.0.0.0/0",
-			GatewayIPAddress: virtualGW,
-		}
-		podIPInfo.Routes = []cns.Route{virtualGWRoute, route}
-	case cns.InfraNIC:
-		// Get and parse infraVNETCIDRs from env
-		infraVNETCIDRs, err := configuration.InfraVNETCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
-		}
-		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
-		}
+	podIPInfos := make([]cns.PodIpInfo, len(mtpnc.Status.InterfaceInfos))
 
-		// Get and parse podCIDRs from env
-		podCIDRs, err := configuration.PodCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get podCIDRs from env")
-		}
-		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podCIDRs")
+	// Set the first element from the first InterfaceInfos
+	if len(mtpnc.Status.InterfaceInfos) > 0 {
+		podIPInfos[0] = cns.PodIpInfo{
+			PodIPConfig: cns.IPSubnet{
+				IPAddress:    ip.String(),
+				PrefixLength: uint8(prefixSize),
+			},
+			MacAddress:        mtpnc.Status.InterfaceInfos[0].MacAddress,
+			NICType:           mtpnc.Status.InterfaceInfos[0].NICType,
+			SkipDefaultRoutes: false,
+			// InterfaceName is empty for DelegatedVMNIC
 		}
+	}
 
-		// Get and parse serviceCIDRs from env
-		serviceCIDRs, err := configuration.ServiceCIDRs()
-		if err != nil {
-			return errors.Wrapf(err, "failed to get serviceCIDRs from env")
-		}
-		serviceCIDRsV4, serviceCIDRsV6, err := utils.ParseCIDRs(serviceCIDRs)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse serviceCIDRs")
+	// Fill rest of the elements from InterfaceInfos
+	for i := 1; i < len(mtpnc.Status.InterfaceInfos); i++ {
+		interfaceInfo := mtpnc.Status.InterfaceInfos[i]
+		podIPInfos[i] = cns.PodIpInfo{
+			PodIPConfig: cns.IPSubnet{
+				IPAddress:    ip.String(),
+				PrefixLength: uint8(prefixSize),
+			},
+			MacAddress:        interfaceInfo.MacAddress,
+			NICType:           interfaceInfo.NICType,
+			SkipDefaultRoutes: false,
+			// InterfaceName is empty for DelegatedVMNIC
```

Comment:
for IB NIC, is interfaceName is not ""? I think we can leave this field here; if DelegatedVMNIC interfaceName is empty, then it will return empty value for this field